examples/rbfe_edge_list.py

```diff
@@ -64,60 +116,26 @@ def read_from_args():
 
-            print(f"Submitting job for {mol_a_name} -> {mol_b_name}")
             mcs_threshold = 2.0
             core, smarts = atom_mapping.get_core_with_alignment(mol_a, mol_b, threshold=mcs_threshold)
-            cfutures.append(
-                cpc.submit(run_complex, mol_a, mol_b, core, forcefield, protein, args.n_frames, args.seed + row_idx)
-            )
-            sfutures.append(
-                cpc.submit(run_solvent, mol_a, mol_b, core, forcefield, protein, args.n_frames, args.seed + row_idx)
-            )
-
-            metadata.append(
-                (
-                    mol_a,
-                    mol_b,
-                    smarts,
-                    core,
-                    float(exp_ddg) * KCAL_TO_KJ,
-                    float(fep_ddg) * KCAL_TO_KJ,
-                    float(fep_ddg_err) * KCAL_TO_KJ,
-                    float(ccc_ddg) * KCAL_TO_KJ,
-                    float(ccc_ddg_err) * KCAL_TO_KJ,
-                )
-            )
 
-    for i, meta in enumerate(metadata):
-        mol_a, mol_b, _, _, exp_ddg, fep_ddg, fep_ddg_err, ccc_ddg, ccc_ddg_err = meta
-        mol_a_name = get_mol_name(mol_a)
-        mol_b_name = get_mol_name(mol_b)
-
-        try:
-            solvent_res, solvent_top = sfutures[i].result()
-            complex_res, complex_top = cfutures[i].result()
-
-            with open(f"success_rbfe_result_{mol_a_name}_{mol_b_name}.pkl", "wb") as fh:
-                pkl_obj = (meta, solvent_res, solvent_top, complex_res, complex_top)
-                pickle.dump(pkl_obj, fh)
-
-            solvent_ddg = np.sum(solvent_res.all_dGs)
-            solvent_ddg_err = np.linalg.norm(solvent_res.all_errs)
-            complex_ddg = np.sum(complex_res.all_dGs)
-            complex_ddg_err = np.linalg.norm(complex_res.all_errs)
-
-            tm_ddg = complex_ddg - solvent_ddg
-            tm_err = np.linalg.norm([complex_ddg_err, solvent_ddg_err])
-
-            print(
-                f"finished: {mol_a_name} -> {mol_b_name} (kJ/mol) | complex {complex_ddg:.2f} +- {complex_ddg_err:.2f} | solvent {solvent_ddg:.2f} +- {solvent_ddg_err:.2f} | tm_pred {tm_ddg:.2f} +- {tm_err:.2f} | exp_ddg {exp_ddg:.2f} | fep_ddg {fep_ddg:.2f} +- {fep_ddg_err:.2f}"
-            )
-
-        except Exception as err:
-            print(
-                f"failed: {err} {mol_a_name} -> {mol_b_name} (kJ/mol) | exp_ddg {exp_ddg:.2f} | fep_ddg {fep_ddg:.2f} +- {fep_ddg_err:.2f}"
+            print(f"Submitting job for {mol_a_name} -> {mol_b_name}")
+            cpc.submit(
+                run_edge_and_save_results,
+                mol_a,
+                mol_b,
+                core,
+                forcefield,
+                protein,
+                args.n_frames,
+                args.seed + row_idx,
+                smarts,
+                exp_ddg,
+                fep_ddg,
+                fep_ddg_err,
+                ccc_ddg,
+                ccc_ddg_err,
             )
```

Comment:
fut.result() blocks until the subjob completes. If the process exits before the subjobs complete, I'm not sure what will happen.

Reply:
In my testing the process has stayed open until subprocesses complete. But maybe best to add an explicit call to `.result()`?

Reply:
Yeah it seems ProcessPoolExecutor will wait, but it might be better to be explicit.

Reply:
Addressed in 7addb66